### ISSUE TITLE
SimpleXML doesn't play nicely with Java17

### ISSUE THREAD

**Issue by jorabin** [jorabin/KeePassJava2#40]
```
[ERROR] Errors: 
[ERROR]   KdbxInnerOutputStreamTest.test:46 » IllegalState java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.util.UUID.mostSigBits accessible: module java.base does not "opens java.util" to unnamed module @a803f94
[ERROR]   SimpleSaveAndReloadTest>SaveAndReloadChecks.saveAndReloadTest:59->saveDatabase:52 » IllegalState java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.util.UUID.mostSigBits accessible: module java.base does not "opens java.util" to unnamed module @a803f94
[ERROR]   SimpleSaveAndReloadTest>SaveAndReloadChecks.saveAndReloadTest2:103->saveDatabase:52 » IllegalState java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.util.UUID.mostSigBits accessible: module java.base does not "opens java.util" to unnamed module @a803f94
[ERROR]   SimpleSaveAndReloadTest.uppercaseBooleanTest:74 » IllegalState java.lang.reflect.InaccessibleObjectException: Unable to make field private final long java.util.UUID.mostSigBits accessible: module java.base does not "opens java.util" to unnamed module @a803f94
```
Looks a lot like it's trying to use reflection and can't.

**Comment by jorabin** [comment]
It doesn't seem as though SimpleXML is under active development (as mentioned under #26), but then again, it may burst into life again (mentioning no names I can think of a project that went quiet for a long time 😄 ), however, for now it looks like it's not a good bet for the future.

Wondering if anyone has some ideas, since as also mentioned (under #30) wondering what kind of place JAXB has in this project going forward? And as also noted elsewhere (in #12 and in the readme, the DOM implementation has a number of virtues, like it doesn't mess up what it doesn't understand) DOM traversal never has been fast.

It's tempting to go with Jackson, but there are other options and wondering if anyone has experience they'd like to share in this area? See also [A Guide to XML in Java](https://www.baeldung.com/java-xml).

**Comment by giusvale-dev** [comment]
We still have many tasks to implement again! See you soon 💪💪💪

**Comment by jorabin** [comment]
I will try to gather up some more issues over coming weeks and will release 2.2.2 to Maven after that.

**Comment by jorabin** [comment]
finally closed as 2.2.2 release
